### PR TITLE
Add option to not trigger update on large files

### PR DESCRIPTION
**Pull request by raven42**
New feature for #636 

Initial addition of large file support option. Just getting the ball rolling on this, but not fully implemented.

Looking to find a way to add output into tagbar window indicating that the file was ignored. Currently this will be displayed once in the vim messages history, but due to ordering of loading, the message isn't actually seen. I believe this is due to the messages history getting initialized after the plugins are loaded.

This adds an option `g:tagbar_file_size_limit` which can be set by the user to set a line count limit so the auto-update does not process anything on the file. This can be overridden by calling a new `tagbar#ForceUpdate()` routine

### CODE REVIEW

File: autoload/tagbar.vim
Comment (alerque) [inline autoload/tagbar.vim:1161]: I like this feature, but in reviewing the code I found it very counter intuitive that the thing being checked is lines, but the setting here is called filesize. File size in my mind is a byte length, not a line count.

One of the most common cases I'd like to exclude is "compiled" source files that have been compressed by removing spaces. These show up a lot in JavaScript work and are usually massive rollups of other code jammed into just a few lines. They would easily be over a reasonable limit for filesize, but not trip a line count limit.

I think either this setting should be renamed to be intuitive or changed so it actually works on filesize. I think I'd prefer the latter. What do you think?

**Review by alerque — CHANGES_REQUESTED** [review]

File: autoload/tagbar.vim
Comment (raven42) [reply autoload/tagbar.vim]: Updated to work on file byte size instead.

**Comment by raven42** [comment]
Any idea how to display some info in the tagbar window for this?

**Review by alerque — DISMISSED** [review]

**Comment by alerque** [comment]
Thanks for working on this @raven42.

> Anybody have any idea how to display some info in the tagbar window for this?

Something would definitely make for a better UX than nothing.

This shouldn't be an error or anything else disruptive to a workflow, but something stuffed in the Tagbar buffer indicating that the file was deliberately not processed would both disambiguate intended from unintended function and give people a clue what to change to get a different result.

**Comment by alerque** [comment]
Just FYI @raven42 I didn't merge this yet because I wasn't sure if you were going to continue on poking at a status report of some kind. If you're not going to get around to that soon feel free to go ahead and merge when you think it's ready. As long as it has at least one current review approval from anybody else with write access you should me able to merge it yourself.

**Comment by raven42** [comment]
@alerque: Thanks, ya I'm not quite ready to merge it in. Was going to try playing around with some way to display some message in the tagbar buffer first. If I can't really come up with anything, then I'll merge this in as is.

**Comment by raven42** [comment]
Updated the code to use the fileinfo so we can track the file size and display a message in the tagbar window in the event the file size limit is exceeded.

**Review by alerque — APPROVED** [review]

**Pull request merged**
